subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -110,11 +119,27 @@ The following classes were moved from `org.gradle.util` to the `org.gradle.util.
 - `FailsWithMessage`
 - `FailsWithMessageExtension`
 
-Also, the unused classes were deleted from `org.gradle.util`:
+The last set of classes have no external or internal usages and therefore were deleted:
 
 - `DiffUtil`
 - `NoopChangeListener`
 - `EnumWithClassBody`
 - `AlwaysTrue`
 - `ReflectionEqualsMatcher`
 - `DynamicDelegate`
+- `TreeVisitor`
```

Comment:
I think this one was moved to internal package and not deleted.

Reply:
Fixed